src/shared/orgShapeListUtils.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { basename, join } from 'path';
+import { AuthInfo, fs, Global, Org, Logger, Connection } from '@salesforce/core';
+
+interface OrgShape {
+  Id: string;
+  CreatedBy: {
+    Username: string;
+  };
+  Status: string;
+  CreatedDate: string;
+}
+
+export interface OrgShapeListResult {
+  orgId: string;
+  username: string;
+  alias?: string;
+  shapeId: string;
+  status: string;
+  createdBy: string;
+  createdDate: string;
+}
+
+export interface JsForceError extends Error {
+  errorCode: string;
+  fields: string[];
+}
+
+export async function getNonScratchOrgs(fileNames: string[]): Promise<AuthInfo[]> {
```

Comment:
If you start this off on coreV3, 1) we won't have to parking-orbit it, 2) GlobalInfo.orgs should have a IsScratch field

Check with Mike to make sure that's reliable for an sfdx plugin, or whether you'd want to go about it some other way.